doc/modules/cassandra/pages/operating/security.adoc

```diff
@@ -43,20 +43,91 @@ not involve changing encryption settings in cassandra.yaml. See
+appropriately. See https://github.com/apache/cassandra/tree/trunk/examples/ssl-factory[ssl-factory examples]
+for details. Refer to the below class diagram to understand the
+class hierarchy,
+
+image::Cassandra-SslContextFactory-PEM.png[SslContextFactory Class Diagram]
+
+=== Using PEM based key material
+
+You can use the inbuilt class `PEMBasedSSLContextFactory` as the
+`ssl_context_factory` setting for the PEM based key material.
+
+You can configure this factory with either inline PEM data or with the
+files having the required PEM data as shown below,
+
+* Configuration: PEM keys/certs defined inline (mind the spaces in the
+YAML!)
+
+....
+   client/server_encryption_options:
+     ssl_context_factory:
+        class_name: org.apache.cassandra.security.PEMBasedSslContextFactory
+        parameters:
+            private_key: |
+             -----BEGIN ENCRYPTED PRIVATE KEY----- OR -----BEGIN PRIVATE KEY-----
+             <your base64 encoded private key>
+             -----END ENCRYPTED PRIVATE KEY----- OR -----END PRIVATE KEY-----
+             -----BEGIN CERTIFICATE-----
+             <your base64 encoded certificate chain>
+             -----END CERTIFICATE-----
+
+            private_key_password: "<your password if the private key is encrypted with a password>"
+
+            trusted_certificates: |
+              -----BEGIN CERTIFICATE-----
+              <your base64 encoded certificate>
+              -----END CERTIFICATE-----
+....
+
+* Configuration: PEM keys/certs defined in files
+....
+    client/server_encryption_options:
+     ssl_context_factory:
+        class_name: org.apache.cassandra.security.PEMBasedSslContextFactory
+     keystore: <file path to the keystore file in the PEM format with the private key and the certificate chain>
+     keystore_password: "<your password if the private key is encrypted with a password>"
+     truststore: <file path to the truststore file in the PEM format>
+....
 
 == SSL Certificate Hot Reloading
 
 Beginning with Cassandra 4, Cassandra supports hot reloading of SSL
-Certificates. If SSL/TLS support is enabled in Cassandra, the node
-periodically polls the Trust and Key Stores specified in cassandra.yaml.
-When the files are updated, Cassandra will reload them and use them for
-subsequent connections. Please note that the Trust & Key Store passwords
-are part of the yaml so the updated files should also use the same
-passwords. The default polling interval is 10 minutes.
+Certificates. If SSL/TLS support is enabled in Cassandra and you are
+using default file based key material, the node periodically polls the
+Trust and Key Stores specified in cassandra.yaml. When the files are
+updated, Cassandra will reload them and use them for subsequent
+connections. Please note that the Trust & Key Store passwords are part
+of the yaml so the updated files should also use the same passwords.
+The default polling interval is 10 minutes.
```

Comment:
how can I change this interval?

Reply:
Good point! Currently this default is not overridable. So we have to change the documentation to make that clearer.

Reply:
I realized that I just picked this from the original documentation where it says 'default polling interval is 10 minutes'. None-the-less we should improve the documentation to be better.

Reply:
@smiklosovic I tried to change wording around that interval since there is no way to configure that. If we do consider making it configurable, we would change this documentation at that time to reflect it accordingly.